packages/next/cli/next-dev.ts

```diff
@@ -7,8 +7,11 @@ import { printAndExit } from '../server/lib/utils'
 import * as Log from '../build/output/log'
 import { startedDevelopmentServer } from '../build/output'
 import { cliCommand } from '../bin/next'
+import detectPort from 'detect-port'
```

Comment:
This should not use an external dependency. You can `.listen()` without a port and read the port after listening.